File: iOS/ssabree/ssabree/Core/Network/CommentService.swift
import Foundation

struct EmptyResponse: Decodable {}

protocol CommentService {
    func likeComment(commentId: Int) async throws -> CommentLikeResponse
    func unlikeComment(commentId: Int) async throws -> CommentLikeResponse
    func updateComment(commentId: Int, content: String) async throws
    func deleteComment(commentId: Int) async throws
}

struct CommentUpdateRequest: Encodable {
    let content: String
}

final class CommentServiceImpl: CommentService {
    private let apiClient: APIClient

    init(apiClient: APIClient = .shared) {
        self.apiClient = apiClient
    }

    func likeComment(commentId: Int) async throws -> CommentLikeResponse {
        let endpoint = APIEndpoint(path: "/api/comments/\(commentId)/like", method: .POST, requiresAuth: true)
        return try await apiClient.request(endpoint: endpoint, body: nil as String?, queryItems: nil)
    }

    func unlikeComment(commentId: Int) async throws -> CommentLikeResponse {
        let endpoint = APIEndpoint(path: "/api/comments/\(commentId)/like", method: .DELETE, requiresAuth: true)
        return try await apiClient.request(endpoint: endpoint, body: nil as String?, queryItems: nil)
    }

    func updateComment(commentId: Int, content: String) async throws {
        let endpoint = APIEndpoint(path: "/api/comments/\(commentId)", method: .PUT, requiresAuth: true)
        let request = CommentUpdateRequest(content: content)
        try await apiClient.requestEmpty(endpoint: endpoint, body: request, queryItems: nil)
    }

    func deleteComment(commentId: Int) async throws {
        let endpoint = APIEndpoint(path: "/api/comments/\(commentId)", method: .DELETE, requiresAuth: true)
        try await apiClient.requestEmpty(endpoint: endpoint, body: nil as String?, queryItems: nil)
    }
}

File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/datasource/remote/CommentService.kt
package com.ssafy.ssabree.core.datasource.remote

import com.ssafy.ssabree.core.datasource.remote.model.CommentLikeResponse
import com.ssafy.ssabree.core.datasource.remote.model.CommentUpdateRequest
import com.ssafy.ssabree.core.datasource.remote.model.CommentResponse
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface CommentService {
    @PUT("/api/comments/{commentId}")
    suspend fun updateComment(
        @Path("commentId") commentId: Long,
        @Body request: CommentUpdateRequest
    ): CommentResponse

    @DELETE("/api/comments/{commentId}")
    suspend fun deleteComment(@Path("commentId") commentId: Long)

    @POST("/api/comments/{commentId}/like")
    suspend fun like(@Path("commentId") commentId: Long): CommentLikeResponse

    @DELETE("/api/comments/{commentId}/like")
    suspend fun unlike(@Path("commentId") commentId: Long): CommentLikeResponse
}
